core-api/src/main/java/com/optimizely/ab/config/audience/match/MatchType.java

```diff
@@ -72,6 +70,16 @@ else if (conditionValue instanceof Boolean) {
         return new MatchType(matchType, new NullMatch());
     }
 
+    private static boolean isValidNumber(Object conditionValue) {
+        if (conditionValue instanceof Integer || conditionValue instanceof Double) {
```

Comment:
If `conditionValue instanceof Integer`, can we also add a restriction that `Math.abs(value) <= 1e53`? Technically it's unnecessary since _Java_ ints can't be that large, but since this is a reference SDK, it might be helpful to set an example for other SDKs.